include/RAJA/policy/openmp_target/reduce.hpp

```diff
@@ -254,17 +268,29 @@ struct TargetReduceLoc
 {
   TargetReduceLoc() = delete;
   TargetReduceLoc(const TargetReduceLoc &) = default;
-  explicit TargetReduceLoc(T init_val, IndexType init_loc)
+  explicit TargetReduceLoc(T init_val, IndexType init_loc,
+                           T identity_ = Reducer::identity)
       : info(),
-        val(Reducer::identity, Reducer::identity, info),
+        val(identity_, identity_, info),
         loc(init_loc, IndexType(RAJA::reduce::detail::DefaultLoc<IndexType>().value()), info),
         initVal(init_val),
-        finalVal(Reducer::identity),
+        finalVal(identity_),
         initLoc(init_loc),
         finalLoc(IndexType(RAJA::reduce::detail::DefaultLoc<IndexType>().value()))
   {
   }
 
+  void reset(T in_val, T identity_ = Reducer::identity)
+  {
+    operator T();
+    initVal = in_val;
+    initLoc = reduce::detail::DefaultLoc<IndexType>().value();
+    finalLoc = IndexType(RAJA::reduce::detail::DefaultLoc<IndexType>().value());
+    finalVal = identity_;
```

Comment:
should this match the order in the constructor for higher intergrokability.

Reply:
Done!